risc0/zkp/src/hal/cuda.rs

```diff
@@ -438,7 +446,19 @@ impl<CH: CudaHash> CudaHal<CH> {
     }
 }
 
-impl<CH: CudaHash> Hal for CudaHal<CH> {
+impl CudaHal<dyn CudaHash> {
+    pub fn new_from_hash_suite(hash_suite: HashSuite<BabyBear>) -> Self {
+        let hash_suite_box = match &hash_suite.name[..] {
+            "poseidon2" => Box::new(CudaHashPoseidon2::new()) as Box<dyn CudaHash>,
+            "poseidon254" => Box::new(CudaHashPoseidon254::new()) as Box<dyn CudaHash>,
+            "sha-256" => Box::new(CudaHashSha256::new()) as Box<dyn CudaHash>,
+            other => unimplemented!("unsupported hash_fn {other}"),
```

Comment:
I think this needs to return an error rather than panic. I'm thinking about whether the hash function might be part of some user-controlled proof request.